library/src/test/java/com/datatorrent/lib/io/fs/FastMergerDecisionMakerTest.java

```diff
@@ -0,0 +1,188 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.lib.io.fs;
+
+import java.io.IOException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.datatorrent.lib.io.fs.FileStitcher.BlockNotFoundException;
+import com.datatorrent.lib.io.fs.HDFSFileMerger.FastMergerDecisionMaker;
+import com.datatorrent.lib.io.fs.Synchronizer.OutputFileMetadata;
+
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link FastMergerDecisionMaker}
+ */
+
+public class FastMergerDecisionMakerTest
+{
+  private static final long DEFAULT_BLOCK_SIZE = 100L;
+  private static final short[] DEFAULT_REPLICATION = new short[] {(short)3, (short)3, (short)3 };
+
+  FastMergerDecisionMaker fastMergerDecisionMaker;
+  @Mock
+  OutputFileMetadata fileMetadata;
+  @Mock
+  FileSystem appFS;
+  @Mock
+  FileStatus status0;
+  @Mock
+  FileStatus status1;
+  @Mock
+  FileStatus status2;
+
+  @Before
+  public void setup()
+  {
+    MockitoAnnotations.initMocks(this);
+
+    when(fileMetadata.getNumberOfBlocks()).thenReturn(3);
+    when(fileMetadata.getBlockIds()).thenReturn(new long[] {0L, 1L, 2L });
+
+    fastMergerDecisionMaker = new FastMergerDecisionMaker("", appFS, DEFAULT_BLOCK_SIZE);
+  }
+
+  /**
+   * If some block is missing then expect BlockNotFoundException.
+   * 
+   * @throws IOException
+   * @throws BlockNotFoundException
+   */
+  @Test(expected = BlockNotFoundException.class)
+  public void testMissingBlock() throws IOException, BlockNotFoundException
+  {
+
+    when(appFS.exists(new Path("/0"))).thenReturn(true);
+    when(appFS.exists(new Path("/1"))).thenReturn(false);
+
+    when(status0.getReplication()).thenReturn(DEFAULT_REPLICATION[0]);
+    when(status0.getLen()).thenReturn(DEFAULT_BLOCK_SIZE);
+
+    when(appFS.getFileStatus(new Path("/0"))).thenReturn(status0);
+
+    fastMergerDecisionMaker.isFastMergePossible(fileMetadata);
+    fail("Failed when one block missing.");
+  }
+
+  private void initializeMocks(long[] blockSizes, short[] replication) throws IOException, BlockNotFoundException
+  {
+    when(appFS.exists(Matchers.any(Path.class))).thenReturn(true);
```

Comment:
you can do it in beforeTest as almost all tests need this. If one of the needs something different, you can override only that part for that particular test.